builtin/providers/aws/resource_aws_ecs_service.go

```diff
@@ -292,7 +292,7 @@ func resourceAwsEcsServiceRead(d *schema.ResourceData, meta interface{}) error {
 	d.Set("desired_count", service.DesiredCount)
 
 	// Save cluster in the same format
-	if strings.HasPrefix(d.Get("cluster").(string), "arn:aws:ecs:") {
+	if regexp.MustCompile(`^arn:[\w-]+:ecs:`).MatchString(d.Get("cluster").(string)) {
```

Comment:
I know I originally suggested using regexp matching here, but now that I'm looking at the whole diff I'm thinking we could just use the same pattern as elsewhere and avoid the loose matching - i.e.

```go
if strings.HasPrefix(d.Get("cluster").(string), "arn:"+meta.(*AWSClient).partition+":ecs:") {
```
or something similar w/ `fmt.Sprintf` - depending on what looks better.

What do you think?